src/journal/journal-file.c

```diff
@@ -464,6 +497,9 @@ static int journal_file_refresh_header(JournalFile *f) {
         /* Sync the online state to disk */
         (void) fsync(f->fd);
 
+        /* We likely just created a new file, also sync the directory this file is located in. */
+        (void) fsync_directory_of_file(f->fd);
```

Comment:
Note that journal_file_open() already takes both an fd and a name these days (though not in openat()-style, because the passed name is the name of the fd). 

I thought about passing the actual directory path, but I didn't want to complicate journal_file_open()'s parameter list even further, it's kinda crazy already... And I figured: a logic like this is useful anyway, if journal_file_open() is called with an fd and no name supplied (after all we have sd_journal_file_open_fd() now), and the fsync() on the dir isn't precisely the most important thing in the world, if it operates on the wrong dir in some exotic corner cases then not too much is lost and nothing bad happens...

That said, I'd be happy to merge a patch that changes this logic around in a later PR?